src/SqlBindingUtilities.cs

```diff
@@ -290,5 +290,51 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Query the ServerProperty('EngineEdition') for logging the Server Type for Telemetry
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Throw if an error occurs while querying the engine edition</exception>
+        public static async Task<string> GetSqlServerEdition(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
```

Comment:
If this is only used for telemetry, then we probably should only be sending these queries if a user hasn't opted out of telemetry. Thoughts?

Reply:
I didn't know users can opt out of telemetry, since this is an overhead call on top of the function  calls, we should skip if users opted out of telemetry.